rule_parser_test.go

```diff
@@ -35,3 +37,57 @@ func TestParseRule(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestMarshalRule(t *testing.T) {
+	err := filepath.Walk("./testdata/", func(path string, info os.FileInfo, err error) error {
+		if !strings.HasSuffix(path, ".rule.yml") {
+			return nil
+		}
+
+		t.Run(strings.TrimSuffix(filepath.Base(path), ".rule.yml"), func(t *testing.T) {
+			contents, err := ioutil.ReadFile(path)
+			if err != nil {
+				t.Fatalf("failed reading test input: %v", err)
+			}
+
+			rule, err := ParseRule(contents)
+			if err != nil {
+				t.Fatalf("error parsing rule: %v", err)
+			}
+
+			// Create a new temporary file in our testing temp directory
+			stream, err := os.CreateTemp(t.TempDir(), filepath.Base(path))
+			if err != nil {
+				t.Fatalf("error creating temp rule file: %v", err)
+			}
+			defer os.Remove(stream.Name())
+			defer stream.Close()
+
+			// Save the rule to a temporary file
+			encoder := yaml.NewEncoder(stream)
+			if err := encoder.Encode(&rule); err != nil {
+				t.Fatalf("error encoding rule to file: %v", err)
+			}
+
+			// Return to the beginning of the stream
+			stream.Seek(0, os.SEEK_SET)
+
+			// Re-read the rule from the newly serialized file
+			var rule_copy Rule
+			decoder := yaml.NewDecoder(stream)
+			if err := decoder.Decode(&rule_copy); err != nil {
+				t.Fatalf("error decoding rule copy: %v", err)
+			}
+
+			if !cmp.Equal(rule, rule_copy) {
+				t.Fatalf("rule and marshalled copy are not equal")
+			}
+
+			cupaloy.New(cupaloy.SnapshotSubdirectory("testdata")).SnapshotT(t, rule)
```

Comment:
```suggestion
```
I don't think there's any need to snapshot here? (assuming this is just leftover from copying `TestParseRule`

Reply:
I'll be totally honest. I don't fully understand the purpose of the snapshotting, and haven't dug very deeply into it. lol So, yes, that's just a bit of copy-pasta from the other unit test. I can remove it if you think it's not necessary. :+1: